### PR TITLE
Refactor mode

### PR DESCRIPTION
**Pull request by halilcengel**

### CODE REVIEW

File: src/state.js [hunk 1]
```diff
@@ -52,6 +52,14 @@ function withSample() {
   return state;
 }
 
+function withPages(nucContext) {
```
Comment (canmingir) [inline src/state.js:55]: Can you shorten this to `{ api, types, functions }`?

File: src/App.jsx [hunk 2]
```diff
@@ -106,60 +123,41 @@ function App() {
...
 
-  const [context, setContext] = React.useState();
+  const InitVfs = (context) => {
+    const files = contextToMap(context.nucleoid);
+    vfs.init(files);
+  };
 
   React.useEffect(() => {
-    async function initContext() {
+    async function InitMode() {
```
Comment (canmingir) [inline src/App.jsx:135]: Can you rename to `initMode`?

File: src/App.jsx [hunk 3]
```diff
@@ -106,60 +123,41 @@ function App() {
...
     }
 
-    const files = contextToMap(context.nucleoid);
-    vfs.init(files);
-
     return context;
   };
 
-  const [context, setContext] = React.useState();
+  const InitVfs = (context) => {
```
Comment (canmingir) [inline src/App.jsx:129]: Can you rename to `initVfs`?

File: src/App.jsx [hunk 4]
```diff
@@ -52,21 +55,35 @@ function App() {
 
   function project(id) {
     return Promise.all([
-      axios.get(`http://localhost:3001/api/services/${id}/context`),
-      axios.get(`http://localhost:3001/api/services/${id}`),
-    ]).then(([contextResult, serviceResult]) => {
-      const context = contextResult.data.context;
+      axios.get(`http://localhost:3000/api/services/${id}/context`),
```
Comment (canmingir) [inline src/App.jsx:58]: Didn't we have to use API from config instead of `localhost:3000`?

**Review by canmingir — CHANGES_REQUESTED** [review]

**Review by durulkoca — APPROVED** [review]

**Review by canmingir — APPROVED** [review]
